core/state/dump.go

```diff
@@ -62,9 +62,10 @@ func (self *StateDB) RawDump() Dump {
 			Code:     common.Bytes2Hex(obj.Code(self.db)),
 			Storage:  make(map[string]string),
 		}
-		storageIt := trie.NewIterator(obj.getTrie(self.db).NodeIterator(nil))
+		so := self.getStateObject(common.BytesToAddress(addr))
+		storageIt := trie.NewIterator(so.getTrie(self.db).NodeIterator(nil))
```

Comment:
please use

```
storage := self.StorageTrie(common.BytesToAddress(addr))
storageIt := trie.NewIterator(storage.NodeIterator(nil))
...
```

Reply:
@fjl your suggestion is a bit incomplete.. Do you mean something like this: 
```golang

		so := self.getStateObject(common.BytesToAddress(addr))
		storage := self.StorageTrie(common.BytesToAddress(addr))
		storageIt := trie.NewIterator(storage.NodeIterator(nil))
		trie := so.getTrie(self.db)
		for storageIt.Next() {
			account.Storage[common.Bytes2Hex(trie.GetKey(storageIt.Key))] = common.Bytes2Hex(storageIt.Value)
		}
		dump.Accounts[common.Bytes2Hex(addr)] = account
```

Reply:
`storage.GetKey` can be used to get the preimage, no need to use `getStateObject` and `getTrie`. I'm suggesting this because it's better to rely on public API so the dumping stuff could be moved out of core/state later.